src/components/charts/ChartGroupUpdate.tsx

```diff
@@ -102,75 +134,213 @@ export default function ChartGroupUpdate({ }) {
         setChartDetailsUpdate(false)
     }
 
-    return (
-        <>
-            <div className="chart-group--details-page">
-                <div className="page-header">
-                    <div className="flex column left">
-                        <div className="flex left">
-                            <BreadCrumb breadcrumbs={breadcrumbs.slice(1,)} />
-                        </div>
-                        <div className="flex left page-header__title">
-                            {state.name}
-                            <Pencil className="pointer" onClick={e => setChartDetailsUpdate(true)} />
-                        </div>
+    function handleAppStoreChange(event): void {
+        event.preventDefault();
+        let searchParams = new URLSearchParams(location.search);
+        let deprecate = searchParams.get(QueryParams.IncludeDeprecated);
+        let chartRepoId = searchParams.get(QueryParams.ChartRepoId);
+        let qs = `${QueryParams.AppStoreName}=${appStoreName}`;
+        if (deprecate) qs = `${qs}&${QueryParams.IncludeDeprecated}=${deprecate}`;
+        if (chartRepoId) qs = `${qs}&${QueryParams.ChartRepoId}=${chartRepoId}`;
+        history.push(`${url}?${qs}`);
+    }
+
+    function handleDeprecateChange(deprecated): void {
+        let searchParams = new URLSearchParams(location.search);
+        let app = searchParams.get(QueryParams.AppStoreName);
+        let chartRepoId = searchParams.get(QueryParams.ChartRepoId);
+        let qs = `${QueryParams.IncludeDeprecated}=${deprecated}`;
+        if (app) qs = `${qs}&${QueryParams.AppStoreName}=${app}`;
+        if (chartRepoId) qs = `${qs}&${QueryParams.ChartRepoId}=${chartRepoId}`
+        history.push(`${url}?${qs}`);
+    }
+
+    function handleCloseFilter() {
+        setSelectedChartRepo(appliedChartRepoFilter)
+    }
+
+    function handleChartRepoChange(selected): void {
```

Comment:
it  seems like handleChartRepoChange, handleDeprecateChange and handleAppStoreChange are similar.. and common method can be extracted

Reply:
clearSearch also

Reply:
yeah it can, fixing